cibuildwheel/linux.py

```diff
@@ -115,25 +125,64 @@ def build(project_dir, package_name, output_dir, test_command, test_requires, be
             gid=os.getgid(),
         )
 
-        docker_process = subprocess.Popen([
+        container_name = 'cibuildwheel-{}'.format(uuid.uuid4())
+
+        command = [
+            'docker',
+            'run',
+            '--env',
+            'CIBUILDWHEEL',
+            '--name', container_name,
+            '-i',
+            '-v', '/:/host',
+            docker_image,
+            '/bin/bash',
+        ]
+        print('docker command: {}'.format(command))
+        docker_process = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            universal_newlines=True,
+        )
+
+        start = monotonic.monotonic()
+        while True:
+            time.sleep(1)
+            command = [
                 'docker',
-                'run',
-                '--env',
-                'CIBUILDWHEEL',
-                '--rm',
-                '-i',
-                '-v', '%s:/project' % os.path.abspath(project_dir),
-                '-v', '%s:/output' % os.path.abspath(output_dir),
-                '-v', '/:/host',
-                docker_image,
-                '/bin/bash'],
-            stdin=subprocess.PIPE, universal_newlines=True)
+                'cp',
+                '{}/.'.format(os.path.abspath(project_dir)),
+                '{}:/project'.format(container_name),
+            ]
+            print('docker command: {}'.format(command))
+            if subprocess.call(command) == 0:
+                break
+
+            if monotonic.monotonic() - start > 60:
+                break
```

Comment:
A comment about why this retry loop is required would be nice :) and perhaps an informative error on timeout